### ISSUE TITLE
Clarification on Encoder Configuration Mismatch Between Training and Inference Scripts

### ISSUE THREAD

**Issue by AdamWang0224** [cvg/depthsplat#65]
Hi, thank you for sharing this great work!

While reviewing the training and inference scripts, I noticed that for the same base model, the encoder configurations differ:

In the inference script, the encoder is set as:

    model.encoder.upsample_factor=4
    model.encoder.lowest_feature_resolution=8
  
Whereas in the training script, the configuration is:

    model.encoder.upsample_factor=2
    model.encoder.lowest_feature_resolution=4

Could you please clarify the relationship between these two configurations? I was trying to reproduce the training process starting from training the depth model under the UniMatch repository, but this discrepancy got me quite confused, as it seems to lead to a dimension mismatch in the DPTHead.

Thanks in advance for your help!

**Comment by haofeixu** [comment]
Hi, thank you for the question. The encoder config depends on the image resolution. For low-resolution images (e.g., 256x256), we choose to use 
```
model.encoder.upsample_factor=2
model.encoder.lowest_feature_resolution=4
```
For high-resolution images (>256x256), we use
```
model.encoder.upsample_factor=4
model.encoder.lowest_feature_resolution=8
```
The detailed description is provided in the "Implementation Details" in Section 4 of our paper.

**Comment by AdamWang0224** [comment]
Thank you so much for your helpful response! Now I make it clear.

As a follow-up, I’ve encountered an issue during training that I hope you could provide some advice on. I am trying to train depthsplat now. Whether I initialize the depthsplat model with my own UniMatch-trained depth model with monocular depth pretrained weights, or directly use the official depthsplat weights, the depth visualization at step 0 of validation appears to be quite good. However, once training begins, the depth prediction quality drops sharply, and although it recovers to some extent after several thousand steps, the overall performance remains noticeably worse than at initialization. I've tried different learning rate and image resolution, as well as some different (near,far) values, but got similar observations.

May I kindly ask if you have any suggestions on where to start debugging this behavior? Or are there any hyperparameters or training settings you would recommend adjusting in such cases?

I’d really appreciate any insights or advice — thanks again for your support and the great work!

**Comment by haofeixu** [comment]
Hi, thank you for the question. I am looking into this and will make an update soon.

**Comment by haofeixu** [comment]
Hi @AdamWang0224 , the code and models have been updated, please pull the latest commit, thanks.
